### PR TITLE
Feat/add certificate id to devices

### PR DESCRIPTION
**Pull request by miguel91it**
Update firmware sdk to work with certificate id instead of device name

### CODE REVIEW

File: examples/esp_idf/certificate_auth/README.md [hunk 1]
```diff
@@ -44,27 +44,27 @@ You should generate a new device certificate for every new device.
...
 cd main/certs
-../../../../../scripts/certificates/generate_device_certificate.sh <project_id> <device_name>
+../../../../../scripts/certificates/generate_device_certificate.sh <project_id> <certificate_id>
 ```
 
 * `project_id` must match the project ID in Golioth.
   This can be found in the Golioth console by clicking on `Projects` and
   using the `Project ID` column.
-* `device_name`: this is the name of the new device. It's a string, and can be anything.
+* `certificate_id`: this is the device's certificate identifier of the new device. It's an immutable string once is set, and can be anything. It also must be unique in the entire project.
```
Comment (sam-golioth) [inline examples/esp_idf/certificate_auth/README.md:53]: Let's wrap long lines

Comment (sam-golioth) [inline examples/esp_idf/certificate_auth/README.md:53]: > be anything

Do we have restrictions on special characters?

Comment (sam-golioth) [inline examples/esp_idf/certificate_auth/README.md:53]:
```suggestion
* `certificate_id`: this is the device's certificate identifier of the new device. It's an immutable string once it is set, and can be anything. It also must be unique in the entire project.
```

File: scripts/certificates/generate_device_certificate.sh [hunk 2]
```diff
@@ -12,23 +12,23 @@ fi
...
-#       1. ${PROJECT_NAME}-${DEVICE_NAME}.key.der - the private key for the device.
+#       1. ${PROJECT_NAME}-${CERTIFICATE_ID}.key.der - the private key for the device.
 #           (should only be known by the device, and kept secret)
-#       2. ${PROJECT_NAME}-${DEVICE_NAME}.crt.der - the public key (certificate) for the device.
+#       2. ${PROJECT_NAME}-${CERTIFICATE_ID}.crt.der - the public key (certificate) for the device.
 #          This is what the device should present to the Golioth server as a client certificate.
 
 PROJECT_NAME=$1
-DEVICE_NAME=$2
+CERTIFICATE_ID=$2 # should be an unique identifier for the device across an entire project
```
Comment (sam-golioth) [inline scripts/certificates/generate_device_certificate.sh:21]:
```suggestion
CERTIFICATE_ID=$2 # should be a unique identifier for the device across an entire project
```

**Review by szczys — APPROVED** [review]

**Pull request merged**